### ISSUE TITLE
Validation loss in progress bar printed line by line (again)

### ISSUE THREAD

**Issue by graille** [Lightning-AI/lightning#16691]
### Bug description

See #330 for details

I also have this issue when using the Running command of Pycharm

![image](https://user-images.githubusercontent.com/7689997/217568856-bb0428d0-e87d-4b7c-b287-0d18f1308aec.png)

TQDM version: 4.64.1
Pytorch Lightning version: 1.9.0

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): LightningModule
#- PyTorch Lightning Version (e.g., 1.5.0): 1.9.0
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0): 
#- Python version (e.g., 3.9): 3.9
#- OS (e.g., Linux): Windows
#- CUDA/cuDNN version: 11.7
#- GPU models and configuration: RTX 3070
#- How you installed Lightning(`conda`, `pip`, source): Conda
#- Running environment of LightningApp (e.g. local, cloud): Local
```

</details>

**Comment by carmocca** [comment]
This is a limitation of tqdm with PyCharm. You can find some suggested options online: https://stackoverflow.com/questions/45012964/stdout-progress-bars-dont-work-in-pycharm

**Comment by astirn** [comment]
How does the training progress bar work fine with PyCharm then?
